nifi-extension-bundles/nifi-atlassian-bundle/nifi-atlassian-extensions/src/main/java/org/apache/nifi/atlassian/bitbucket/BitbucketRepositoryClient.java

```diff
@@ -334,34 +500,249 @@ public InputStream deleteContent(final String filePath, final String commitMessa
+                    String.format("Error while deleting content for repository [%s] on branch %s at path %s: %s", repoName, branch, resolvedPath, getErrorMessage(response)));
+        }
     }
 
     private Iterator<JsonNode> getFiles(final String branch, final String resolvedPath) throws FlowRegistryException {
         final Optional<String> lastCommit = getLatestCommit(branch, resolvedPath);
 
         if (lastCommit.isEmpty()) {
-            throw new FlowRegistryException(String.format("Could not find committed files at %s on branch %s response from Bitbucket API", resolvedPath, branch));
+            return Collections.emptyIterator();
         }
 
-        // retrieve source data
-        // https://api.bitbucket.org/2.0/repositories/{workspace}/{repoName}/src/{commit}/{path}
-        final URI uri = getUriBuilder().addPathSegment("src").addPathSegment(lastCommit.get()).addPathSegment(resolvedPath).build();
+        return formFactor == BitbucketFormFactor.DATA_CENTER
+                ? getFilesDataCenter(branch, resolvedPath, lastCommit.get())
+                : getFilesCloud(branch, resolvedPath, lastCommit.get());
+    }
+
+    private Iterator<JsonNode> getFilesCloud(final String branch, final String resolvedPath, final String commit) throws FlowRegistryException {
+        final URI uri = getRepositoryUriBuilder()
+                .addPathSegment("src")
+                .addPathSegment(commit)
+                .addPathSegment(resolvedPath)
+                .build();
         final String errorMessage = String.format("Error while listing content for repository [%s] on branch %s at path %s", repoName, branch, resolvedPath);
 
         return getPagedResponseValues(uri, errorMessage);
     }
 
+    private Iterator<JsonNode> getFilesDataCenter(final String branch, final String resolvedPath, final String commit) throws FlowRegistryException {
+        final List<JsonNode> allValues = new ArrayList<>();
+        Integer nextPageStart = null;
+
+        do {
+            final URI uri = buildBrowseUri(resolvedPath, commit, nextPageStart, false);
+            final HttpResponseEntity response = this.webClient.getWebClientService()
+                    .get()
+                    .uri(uri)
+                    .header(AUTHORIZATION_HEADER, authToken.getAuthzHeaderValue())
+                    .retrieve();
+
+            if (response.statusCode() != HttpURLConnection.HTTP_OK) {
+                final String errorMessage = String.format("Error while listing content for repository [%s] on branch %s at path %s", repoName, branch, resolvedPath);
+                throw new FlowRegistryException(errorMessage + ": " + getErrorMessage(response));
+            }
+
+            final JsonNode root;
+            try {
+                root = objectMapper.readTree(response.body());
+            } catch (IOException e) {
+                throw new FlowRegistryException(String.format("Could not parse Bitbucket API response at %s", uri), e);
+            }
```

Comment:
This seems to be a common pattern, it would be helpful to create a shared method and reuse instead of copy the try-catch block across methods.